include/zephyr/net/wifi_mgmt.h

```diff
@@ -415,6 +422,21 @@ struct wifi_raw_scan_result {
 	uint8_t data[CONFIG_WIFI_MGMT_RAW_SCAN_RESULT_LENGTH];
 };
 #endif /* CONFIG_WIFI_MGMT_RAW_SCAN_RESULTS */
+
+/** Wi-Fi promiscuous mode setup */
+struct wifi_promisc_setup {
+	union {
+		/** promiscuous mode settings */
+		enum wifi_promiscuous_mode mode;
+		/** filter settings for specific promiscuous mode */
+		enum wifi_promiscuous_filter filter;
```

Comment:
Given that filters can be combined, I don't think it's correct for this variable to be of `enum wifi_promiscious_filter`  type? We could provide a value here that is not specified by the enumeration (`WIFI_PROMISC_MGMT | WIFI_PROMISC_DATA` for instance). Should it be `uint8_t` instead?

Reply:
I agree. I missed this point. Let me re-work this a bit. Will update the code

Reply:
For the statement:
"A follow up question then - wouldn't it make sense then for WIFI_PROMISC_ALL to be a bitwise sum of all of the filter types (WIFI_PROMISC_MGMT | WIFI_PROMISC_DATA | WIFI_PROMISC_CTRL)? In theory, it should have the same effect, so I don't see a reason to differentiate here."

A separate filter setting is given to allow the lower layer not to be bothered by future changes or additions to filter settings. Say, tomorrow, we decide to add action frames are only sniffed, or say, beacon frames are only sniffed etc. Then the definition of WIFI_PROMISC_ALL keeps changing for the lower layer. if a specific filter setting takes the notion of "All packets are sniffed and sent up the stack", the lower layer does not need to be worried about changes to the definition of all

Reply:
ok, fair point.

Reply:
If per se, ALL filter setting is mixed with other individual setting, the lower layer can suitably respond with an error statement